brdnet/subset_studies.py

```diff
@@ -0,0 +1,284 @@
+
+        error = abs(plier_target - plier_sample_count)
+
+        # We have to cast these as lists because python hates it when you modify a dictionary
+        # as you are iterating over it
+        plier_study_list = list(plier_study_counter.keys())
+        for study in plier_study_list:
+            # If we already have too few samples from plier, don't look to move any over
+            plier_sample_count = sum(plier_study_counter.values())
+            error = abs(plier_target - plier_sample_count)
+            if plier_sample_count < plier_target:
+                break
+
+            if abs((plier_sample_count - plier_study_counter[study]) - plier_target) < error:
+                # Move study from plier_study_counter to classifier_study_counter
+                classifier_study_counter[study] = plier_study_counter[study]
+                del plier_study_counter[study]
+
+        classifier_study_list = list(classifier_study_counter.keys())
+        for study in classifier_study_list:
+            # If we already have too few samples from classifier, don't look to move any over
+            classifier_sample_count = sum(classifier_study_counter.values())
+            plier_sample_count = sum(plier_study_counter.values())
+            error = abs(plier_target - plier_sample_count)
+            if classifier_sample_count < classifier_target:
+                break
+
+            error_if_moved = abs((classifier_sample_count - classifier_study_counter[study]) -
+                                 classifier_target)
+            if error_if_moved < error:
+                # Move study from plier_study_counter to classifier_study_counter
+                plier_study_counter[study] = classifier_study_counter[study]
+                del classifier_study_counter[study]
+
+        # If we didn't move any samples between the two sets then we're done balancing
+        if old_plier_count == plier_sample_count:
+            done_balancing = True
+
+    plier_study_list = plier_study_counter.keys()
+    classifier_study_list = classifier_study_counter.keys()
+
+    return plier_study_list, classifier_study_list
+
+
+def get_df_subset_by_study(df, studies_to_keep):
+    '''Create a dataframe containing only the samples that are from a study in studies_to_keep
+
+    Arguments
+    ---------
+    df: pandas.DataFrame
+        The dataframe to subset
+    studies_to_keep: list of str
+        The list of SRA study ids (e.g. SRP012345) to keep
+
+    Returns
+    -------
+    filtered_samples: pandas.DataFrame
+        The subset of df to keep
+    '''
+    # TODO studies_to_keep is zero, find out why
+    studies_regex = '|'.join(studies_to_keep)
+    filtered_samples = df.filter(regex=studies_regex)
+    return filtered_samples
+
+
+def rebalance(plier_healthy, plier_disease, classifier_healthy, classifier_disease, fraction=.2):
+    '''Ensure that the dataframe to be put into PLIER and the dataframe to be used in
```

Comment:
What's the difference between putting data into PLIER and into the classifier here? I thought you were using the compressed PLIER output as input to your classifier - am I missing something?

Reply:
That's a very good question. The way I currently have the pipeline set up, PLIER is trained on a different portion of the data than the classifier is. The trained Z matrix is then used to perform the same transformation on the other portion of the data before the classifier is trained. 

Since the goal is to ultimately use the classifier on diseases we don't have in the training set, I wanted to ensure that there wouldn't be information leakage in the PLIER transformation by having the classifier train on studies and diseases that PLIER knows nothing about.